### DISCUSSION TITLE
apikey в env переменной в Nuxt3

### DISCUSSION THREAD

**Discussion by secam** [yandex-maps-unofficial/vue-yandex-maps#303] [Q&A (2.x)]
В  Nuxt3 не получается положить ключик в env переменную, он всегда хочет, чтобы ключ был прописан явно в настройке модуля
```  
modules: [
    'vue-yandex-maps/nuxt',
  ],
  yandexMaps: {
    apikey: 'xxxx-xxx-xxx'
  }
```
Попытки положить его в `runtimeConfig.public.yandexMaps`, или как-то явно вызвать createYmapsOptions где-то в компонентах или app.ts успехом не увенчались
```
import { createYmapsOptions, VueYandexMaps } from 'vue-yandex-maps'
const config = useRuntimeConfig()

if (!VueYandexMaps.settings.value.apikey) {
  createYmapsOptions({ apikey: config.public.yandexMaps});
}

```
Получаю 500 ошибку при попытке открыть любую страницу nuxt сайта
```
err: {
      "type": "H3Error",
      "message": "You must specify apikey for createYmapsOptions: You must specify apikey for createYmapsOptions",
      "stack":
          YandexMapException: You must specify apikey for createYmapsOptions

```

**Comment by daniluk4000** [comment]
Приветствую! Да, действительно, оно ломается при запуске. У вас есть несколько вариантов:

1. Используйте process.env.YOUR_API_TOKEN в конфиге. Для того, чтобы это работало на продуктиве, переменная должна присутствовать при сборке проекта. Мы так делаем на нашем проекте и в целом живы
2. Положите туда рандомную строку и в плагине измените VueYandexMaps.settings.value.apikey
